### PR TITLE
Fixes 2307

### PR DESCRIPTION
**Pull request by harshgMSFT**

### CODE REVIEW

**Comment by dougbu** [comment]
:shipit: but update commit description to include a detail or two e.g. what's changing.  also reference the correct issue -- #2307.  2370 doesn't (yet) exist.

**Comment by rynowak** [comment]
Please update https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.ModelBinding/ModelMetadata.cs#L255 to cache the result, that's the point of the bug :laughing: 

`ModelMetadata` instance are cached, so doing it in a private field is fine.

**Comment by harshgMSFT** [comment]
@rynowak updated.

**Comment by Eilon** [comment]
:shipit: from me. Also please update the PR title so it's easier to find in the history.

File: src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ public class MutableObjectModelBinder : IModelBinder
         public virtual async Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContext)
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.ModelBinding/Binders/MutableObjectModelBinder.cs]: Unrelated, but mark this as `[NotNull]`

**Comment by pranavkm** [comment]
:shipit:

**Comment by rynowak** [comment]
:shipit:

**Comment by harshgMSFT** [comment]
0c0fa10ed0687e5e7885c4bda147858bc486ac05

**Pull request closed without merging**